Assets/Scripts/Projectile.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Projectile : MonoBehaviour {
+	public float xVel;
+	public bool isRight;
+	float yVel = 0.0f;
+	Rigidbody2D rb;
+
+	void Start() {
+		rb = GetComponent<Rigidbody2D>();
+	}
+
+	void Update() {
+		if (isRight) {
+			rb.velocity = new Vector2(xVel, yVel);
+		} else {
+			rb.velocity = new Vector2(-1 * xVel, yVel);
+		}
+	}
+
+	private void OnCollisionEnter2D(Collision2D other) {
+        if (other.collider.tag == "Player") {
+            PlayerAvatar player = other.gameObject.GetComponent<PlayerAvatar>();
+
+            player.TakeDamage(1, this.gameObject);
```

Comment:
The reason I made `TakeDamage` have a source was so that I could record who got the kill (e.g. player kill, self kill, environment kill)

So perhaps this projectile could have an optional "owner", and we can either (1) pass that owner in directly or (2) have the `TakeDamage` function handle it... "if it's a projectile, check if that projectile has an owner.. assign the kill to them"

https://github.com/CleverGameDev/Cleverfall/blob/3b7322b55a18129a515b06128723be6eb9bc10a1/Assets/Scripts/PlayerAvatar.cs#L104-L108